gammapy/modeling/models/spectral.py

```diff
@@ -213,6 +213,31 @@ def evaluate_error(self, energy, epsilon=1e-4):
         """
         return self._propagate_error(epsilon=epsilon, fct=self, energy=energy)
 
+    def pivot_energy(self):
+        """The decorrelation energy for a given spectral model calculated numerically.
+
+        Returns
+        -------
+        pivot energy : `~astropy.units.Quantity`
+            The energy at which the statistical error is smallest.
+        """
+
+        x_unit = u.GeV
+
+        def min_energy(x):
+            x = np.exp(x)
+            dnde, dnde_error = self.evaluate_error(x * x_unit)
+            energy_minimum = dnde_error / dnde
+            return energy_minimum
+
+        bounds = np.log([self.reference.value - 5, self.reference.value + 5])
+        minimizer = scipy.optimize.minimize_scalar(min_energy, bounds=bounds)
+
+        if not minimizer.success:
+            return print("Minimizer terminated unsuccessfully.")
```

Comment:
In one case, the function return a string and in the other a Quantity. 
I think that it is best that in this case:
- there is a log.warning message (that will precise from which function the message is coming from)
- and return a nan * x_unit